### DISCUSSION TITLE
LSP for vector index feature

### DISCUSSION THREAD

**Discussion by ShadowMitia** [clangd/clangd#1460] [Q&A] [answered]
Hello!

I'm trying to implement LSP inlay hints in Emacs using Eglot.

I'm slowly getting, but I'm confused about different results between what I see in VSCode and Emacs.
Overall I see more hints in VSCode, but especially I see it can show indices for std::vector for example?

VSCode:

![image](https://user-images.githubusercontent.com/3752363/211674072-c9a6e26f-ce44-4ea1-9e3f-db6539cf754b.png)

Emacs

![image](https://user-images.githubusercontent.com/3752363/211674133-c3d13d14-9edc-4119-98d1-4c230d7e00ec.png)

Does anyone know if this is really the inlay hint feature, or is it another LSP feature?

Thanks!

**Comment by sam-mccall** [comment, marked as answer]
Are you using the same version/config of clangd in both cases? The array indices are "designator" inlay hints, and were added relatively recently.

**Reply by ShadowMitia** [reply]
So this would be the same "hint" at work? 
![image](https://user-images.githubusercontent.com/3752363/211733299-e267679a-f125-467e-94f9-d19da37a8354.png)

**Reply by HighCommander4** [reply]
Yep, those are also designator hints

**Reply by ShadowMitia** [reply]
I'm using the same version for both. I just wasn't sure if it was some kind of extension. It's probably something wrong on my side. Thanks!
